torch/backends/cudnn/__init__.py

```diff
@@ -17,6 +17,11 @@ def _libcudnn():
         if hasattr(lib, 'cudnnGetErrorString'):
             lib.cudnnGetErrorString.restype = ctypes.c_char_p
             __cudnn_version = lib.cudnnGetVersion()
+            compile_version = torch._C._cudnn_version()
+            if __cudnn_version != compile_version:
+                raise RuntimeError(
+                    'cuDNN version mismatch: PyTorch was compiled against {} '
+                    'but linked against {}'.format(compile_version, __cudnn_version))
```

Comment:
I was tempted to require that they match exactly. Should we instead check that:

`(__cudnn_version // 100) == (compile_version // 100)`

(That the major + minor match, but ignore patch level)